src/test/kotlin/app/batch/S3ReaderTest.kt

```diff
@@ -55,17 +52,14 @@ class S3ReaderTest {
     private lateinit var s3Object1: S3Object
     private lateinit var s3Object2: S3Object
 
-    @Autowired
-    private lateinit var s3Reader: S3Reader
-
     @MockBean
     private lateinit var s3Client: AmazonS3
 
-    @MockBean
-    private lateinit var httpClientProvider: HttpClientProvider
+    @Autowired
+    private lateinit var keyPairGenerator: KeyPairGenerator
 
-    @MockBean
-    private lateinit var connection: Connection
+    @Autowired
+    private lateinit var s3Reader: S3Reader
```

Comment:
should_Throw_Exception_When_Metadata_File_Missing_In_A_Given_Prefix - can this please assert the exception message, so we can be sure it is both useful and does not leak data?

Reply:
S3 Reader delegates all the logic to keypair  generator . All logging assertions have been done in its test keypairgenerator test